### ISSUE TITLE
Technical doubt regarding Logistic regression

### ISSUE THREAD

**Issue by mayank2794** [aihubprojects/Logistic-Regression-From-Scratch-Python#1]
I am Mayank Agrawal from India, studying Machine Learning from your book titled "Essential Math for Data Science" by Thomas Nield. I was studying Logistic Regression topic and I am facing technical problem when I am writing code for the same in python from scratch.

Example referred from book is patient data given in Example 6-8.
Python-Code is attached herewith for reference along with Input data excel file. I have used the matrix form of Gradient descent. but my answer is coming wrong as I am decreasing the learning rate.
The learning rate used in your example is 0.01, but my answer is coming correct only when learning rate is 0.1, else not. If I am decresing the learning rate to .01 answer is coming wrong.
Can you please help me regarding this.

Regards,
Mayank agrawal
India

# -*- coding: utf-8 -*-
"""
Created on Sun Jul  9 17:13:45 2023

@author: Mayank
"""

import sys
import pandas as pd
import matplotlib
import numpy as np
import scipy as sp
import IPython
import sklearn
from sklearn.linear_model import LogisticRegression

patient_data = pd.read_excel(r'C:\Users\mayank.agrawal\Downloads\patient_data.xlsx', sheet_name='patient_data')
p_d=np.asarray(patient_data)
n=len(patient_data)   #No. of Data Points
x=p_d[:,0:1]
y=p_d[:,1:2]
m=x.shape[1]+1  #No. of features
X=np.c_[np.ones(n),x] #Add a columns of ones
a=np.zeros((m,1)) # random initialization
a[0,0]=1
a[1,0]=1
lr=.1
grad=np.zeros((m,1))
def sigmoid(p):
    sig=1/(1+np.exp(p))
    return sig
for k  in range(10000):
    s=np.matmul(X,a)
    sigma=sigmoid(s)
    grad=-1/n*np.dot(np.transpose(X),sigma-y)
    a=a-lr*grad
print(a)

# Check using SKlearn          
# Perform logistic regression
# Turn off penalty
model = LogisticRegression(penalty='none')
model.fit(x, y)
# print beta1
print(model.coef_.flatten()) # 0.69267212
# print beta0
print(model.intercept_.flatten()) # -3.17576395

    

[patient_data.xlsx](https://github.com/aihubprojects/Logistic-Regression-From-Scratch-Python/files/12014794/patient_data.xlsx)
